controllers/sandbox_controller.go

```diff
@@ -70,58 +75,119 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
-	// Check if the pod already exists, if not create a new one
-	podInCluster := &corev1.Pod{}
-	found := true
-	err := r.Get(ctx, types.NamespacedName{Name: sandbox.Name, Namespace: sandbox.Namespace}, podInCluster)
+	_, err := r.reconcilePod(ctx, sandbox)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = r.reconcileService(ctx, sandbox)
 	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Update the sandbox status
+	// TODO: Update the sandbox status
+
+	return ctrl.Result{}, nil
+}
+
+func (r *SandboxReconciler) reconcileService(ctx context.Context, sandbox *sandboxv1alpha1.Sandbox) (*corev1.Service, error) {
+	log := log.FromContext(ctx)
+	service := &corev1.Service{}
+	found := true
+	if err := r.Get(ctx, types.NamespacedName{Name: sandbox.Name, Namespace: sandbox.Namespace}, service); err != nil {
 		if !errors.IsNotFound(err) {
-			log.Error(err, "Failed to get Pod")
-			return ctrl.Result{}, err
+			log.Error(err, "Failed to get Service")
+			return nil, err
 		}
 		found = false
 	}
 
-	// Create a pod object from the sandbox
-	pod, err := r.podForSandbox(sandbox)
-	if err != nil {
-		return ctrl.Result{}, err
+	if found {
+		log.Info("Found Service", "Service.Namespace", service.Namespace, "Service.Name", service.Name)
+		return service, nil
 	}
-	if !found {
-		log.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		if err = r.Create(ctx, pod, client.FieldOwner("sandbox-controller")); err != nil {
-			log.Error(err, "Failed to create", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-			return ctrl.Result{}, err
+
+	log.Info("Creating a new Headless Service")
+	labelValue := sandbox.Name
+	if len(labelValue) > 63 {
+		labelValue = labelValue[:63]
```

Comment:
This will cause sandboxes with name longer than 63 and the same prefix to select the same pods 
Suggest using a safer way to uniquely filtering resources. For example, can you hash the name?

Reply:
fixed using the same hash as used by replication controller